components/organisms/BookChildren.tsx

```diff
@@ -54,81 +51,78 @@ type Props = {
-  onItemEditClick?(event: MouseEvent<HTMLElement>, index: ItemIndex): void;
+  onItemClick(index: ItemIndex): void;
+  onItemEditClick?(index: ItemIndex): void;
 };
 
-export default function BookChildren(props: Props) {
-  const {
-    className,
-    sections,
-    index: [sectionIndex, topicIndex],
-    isTopicEditable,
-    onItemClick,
-    onItemEditClick,
-  } = props;
+export default function BookChildren({
+  className,
+  sections,
+  index: [sectionIndex, topicIndex],
+  isTopicEditable,
+  onItemClick,
+  onItemEditClick,
+}: Props) {
   const classes = useStyles();
-  const [open, setOpen] = useState<boolean[]>(sections.map(() => true));
-  const handleItemClick = (event: MouseEvent<HTMLElement>) => {
-    const { section, topic } = event.currentTarget.dataset;
-    onItemClick(event, ([section, topic].map(Number) as unknown) as ItemIndex);
-  };
-  const handleSectionClick = (sectionItemIndex: number) => () => {
-    setOpen((open) => {
-      const newOpen = open.slice();
-      newOpen[sectionItemIndex] = !newOpen[sectionItemIndex];
-      return newOpen;
-    });
-  };
-  const handleItemEditClick = (...index: ItemIndex) => (
+  const handleItemClick = (index: ItemIndex) => () => onItemClick(index);
+  const handleSectionItemClick = (sectionItemIndex: number) => () =>
+    onItemClick([sectionItemIndex, 0]);
+  const handleItemEditClick = (index: ItemIndex) => (
     event: MouseEvent<HTMLButtonElement>
   ) => {
     event.stopPropagation();
-    onItemEditClick?.(event, index);
+    onItemEditClick?.(index);
   };
   return (
     <List disablePadding className={className}>
       {sections.map((section, sectionItemIndex) => (
-        <Section
+        <SectionItem
           key={section.id}
           section={section}
           sectionItemIndex={sectionItemIndex}
-          onSectionClick={handleSectionClick(sectionItemIndex)}
-          open={open[sectionItemIndex]}
+          end={sectionItemIndex === sections.length - 1}
+          onSectionItemClick={handleSectionItemClick(sectionItemIndex)}
         >
           {section.topics.map((topic, topicItemIndex) => (
-            <ListItem
+            <BookChildrenItem
               key={`${topic.id}:${topicItemIndex}`}
               className={clsx({
                 [classes.active]:
                   sectionIndex === sectionItemIndex &&
                   topicIndex === topicItemIndex,
               })}
+              variant="topic"
+              outlineNumber={getOutlineNumber(
+                section,
+                sectionItemIndex,
+                topicItemIndex
+              )}
+              name={topic.name}
+              end={
+                (section.topics.length > 1 &&
+                  topicItemIndex === section.topics.length - 1) ||
+                (section.topics.length === 1 &&
+                  sectionItemIndex === sections.length - 1)
+              }
+              depth={section.topics.length > 1 ? 1 : 0}
```

Comment:
こっちのほうが好きですが、これは私の好みの問題。

```suggestion
              depth={Math.min(1, section.topics.length)}
```

そもそもここでdepthが1であることはL86から明らかなように思った。
よって下記で済むのではないでしょうか。

```suggestion
              depth={1}
```

Reply:
章番号の詳細度(xなのか、x.xなのか)に応じて字下げがされてほしいのが実装時の意図でした。

![image](https://user-images.githubusercontent.com/9744580/125236763-76b50c00-e31f-11eb-8784-d19c5300b3d9.png)

`depth={1}` に変更すると `2 リンゴに夢中のレッサーパンダ` の字下げ分の差異が生じます。

![image](https://user-images.githubusercontent.com/9744580/125236825-93514400-e31f-11eb-94c8-7b8cb9d1a3f9.png)

Reply:
なるほど! 読み間違えてしまって意図を勘違いしてしまいました 💦 すみません。
つまりあえて書き換えるならばこういうことですね。

```suggestion
              depth={Math.min(1, section.topics.length - 1)}
```

あるいは `section.topics.length` に対して従属な意図を表明するような変数に置くなどしたほうが良いかも知れないですね (「詳細度」ならばspecificityなどでしょうか)。

Reply:
もし仮に Math.min() を使うならば:

```suggestion
              depth={Math.min(1, section.topics.length - 1)}
```

Reply:
`depth` は1以上も想定していて実質的には字下げの幅がdepthの値に比例するのですが、現状実際に使用するdepthの値は0か1だけですね。

> `section.topics.length` に対して従属な意図を表明するような変数に置くなどするとよりよいかも知れないですね (「詳細度」ならばspecificityなどでしょうか)。

意図している実装、条件に用いている `section.topics.length` 、プロパティ名のdepthのそれぞれの関連性が薄い名前付けであることが改善されてほしい、というニュアンスであっていますでしょうか？

わたしも変更案考えてみます

> もし仮に Math.min() を使うならば:

こちらの提案を反映したいですがそれをするとresolveされるのと名前付けが解決されないのでひとまず置いておきますね！

Reply:
> 意図している実装、条件に用いている `section.topics.length` 、プロパティ名のdepthのそれぞれの関連性が薄い名前付けであることが改善されてほしい、というニュアンスであっていますでしょうか？

はい。`section.topics.length > 1` がコメントにもある「章番号の詳細度」であるということを表明してあるとより良いのではないかということですね。どちらでも構いません。式 `section.topics.length > 1 ? 1 : 0` の振る舞いそのものに関しては特に言うことはありません。

Reply:
既存のコードで用いているisNamedSectionという関数が使えることを失念していました…
章番号の詳細度という概念自体は今までコードとして表現していなかったのですが、
「名前つきのセクション」かどうかで章番号の出力とBookChildrenのリスト表示のロジックを制御していました

https://github.com/npocccties/chibichilo/blob/96f22b1ff738fb49bba5928b5c1ea3b563c1bc4c/utils/outline.ts#L4-L8

dd49013fc75d6935d4bcc1bd592459b334b3dc6a にて上記関数に差し替えるよう反映したので、改めて説明変数が必要など指摘いただければそのように反映したいと思います